### DISCUSSION TITLE
LoaderError "public/theme/base-2021\./" directory does not exist

### DISCUSSION THREAD

**Discussion by Dok11** [bolt/core#2392] [General]
I pass steps from installation guide (https://docs.bolt.cm/4.0/installation/installation) and after `bin/console server:start` I see LoaderError "The "/public/theme/base-2021\./" directory does not exist ("/side-by-side/public/theme/base-2021\./")."

```
Twig\Error\LoaderError:
The "D:/OSPanel/domains/side-by-side/public/theme/base-2021\./" directory does not exist ("D:/OSPanel/domains/side-by-side/public/theme/base-2021\./").

  at D:\OSPanel\domains\side-by-side\vendor\twig\twig\src\Loader\FilesystemLoader.php:108
  at Twig\Loader\FilesystemLoader->prependPath('D:/OSPanel/domains/side-by-side/public/theme/base-2021\\./', '__main__')
     (D:\OSPanel\domains\side-by-side\vendor\bolt\core\src\Controller\TwigAwareController.php:198)
  at Bolt\Controller\TwigAwareController->setTwigLoader()
     (D:\OSPanel\domains\side-by-side\vendor\bolt\core\src\Controller\TwigAwareController.php:93)
  at Bolt\Controller\TwigAwareController->render(array('index.twig', 'index.html.twig'), array('record' => object(Content), 'homepage' => object(Content), 'user' => null, 'theme' => object(DeepCollection)))
     (D:\OSPanel\domains\side-by-side\vendor\bolt\core\src\Controller\TwigAwareController.php:161)
  at Bolt\Controller\TwigAwareController->renderSingle(object(Content), false, array('index.twig', 'index.html.twig'))
     (D:\OSPanel\domains\side-by-side\vendor\bolt\core\src\Controller\Frontend\HomepageController.php:56)
  at Bolt\Controller\Frontend\HomepageController->homepage(object(ContentRepository))
     (D:\OSPanel\domains\side-by-side\vendor\symfony\http-kernel\HttpKernel.php:157)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (D:\OSPanel\domains\side-by-side\vendor\symfony\http-kernel\HttpKernel.php:79)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (D:\OSPanel\domains\side-by-side\vendor\symfony\http-kernel\Kernel.php:195)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (D:\OSPanel\domains\side-by-side\public\index.php:39)  
```

![image](https://user-images.githubusercontent.com/2697890/107835843-ed1f0380-6dab-11eb-84d7-4a66414235c2.png)

The folder "public" does not contain "theme".
Looks like I passed some step or some command didnt work correctly.
After which step I was should get "theme" folder?

![image](https://user-images.githubusercontent.com/2697890/107835971-759da400-6dac-11eb-847c-41b6f98b63b5.png)

```
$ php bin/console bolt:info

⚙️  Bolt
========

 Bolt version: 4.1.12

 * Install type: Packaged distribution
 * Database: mysql 5.5.5-10.3.13-MariaDB - localhost via TCP/IP (with JSON)
 * PHP version: 7.2.10
 * Symfony version: v5.2.1
 * Operating System: Windows NT - 10.0
```

**Comment by bobdenotter** [comment]
Somehow, thi installation missed the step to install the templates. Try running this: 

```
bin/console bolt:copy-themes
```

If that doesn't work, what output does it give? Does the `public/themes/` folder exist, and does it have files in it?

**Reply by Dok11** [reply]
Thank you! This command fix the bug to me, page succesfully open and have no errors in php
But some images from page cant be open
![image](https://user-images.githubusercontent.com/2697890/107855350-c5bf4980-6e32-11eb-89cd-d34ebc56af01.png)

As I found the page require gdlib which required to works. It mentioned in the Required section but is it correctly to hide warnings about missing required extensions before setup works?

And one question: does exist way to autocheck my installation?

**Reply by bobdenotter** [reply]
It does seem like it's missing gdlib, indeed.. Open one of those `/thumbs/…` in your browser, and it'll tell you what the actual error is. That should help fixing it one step more

**Reply by Dok11** [reply]
New error when I try to open /bolt in the browser: `An exception has been thrown during the rendering of a template ("Could not find the entrypoints file from Webpack: the file "D:\OSPanel\domains\side-by-side/public/assets/entrypoints.json" does not exist.").`

The assets folder does not exist. How to get it?

**Reply by Dok11** [reply]
Oh, Im sorry `bin/console bolt:copy-assets` fixed it =)

**Comment by credomane** [comment]
I've ran into everything in this issue. Looking over the different 4.x trees at no point does it look like the `bin/console bolt:setup` runs the command `copy-themes` but at some point `copy-assets` command is ran as the assets are there. Thus leading to a broken install immediately.

Looking over the 4.0 install guide the individual install steps don't list `bin/console copy-themes` as a required step or even as optional step if you were to make your own theme instead.

Also `bin/console bolt:reset-secret` is missing from the individual commands in the 4.0 docs. the all-in-one bolt:setup calls it.

**Comment by mind-overflow** [comment]
Running into exactly the same issue. Should be a public issue instead of a discussion imho.
